syntax = "proto3";

package product;

message Product {
    int32 id = 1;
    string name = 2;
    string description = 3;
    int32 price = 4;
    string image_url = 6;
    int32 stock = 7;
}

message ProductListRequest {
}

message ProductListResponse {
    repeated Product products = 1;
}

message ProductRequest {
    int32 id = 1;
}

message ProductResponse {
    Product product = 1;
}

message ProductCreateRequest {
    string name = 1;
    string description = 2;
    int32 price = 3;
    string image_url = 4;
    int32 stock = 5;
}

message ProductCreateResponse {
    Product product = 1;
}

message ProductUpdateRequest {
    int32 id = 1;
    string name = 2;
    string description = 3;
    int32 price = 4;
    string image_url = 5;
    int32 stock = 6;
}

message ProductUpdateResponse {
    Product product = 1;
}

message ProductDeleteRequest {
    int32 id = 1;
}

message ProductDeleteResponse {
    string message = 1;
}

service ProductService {
    rpc List(ProductListRequest) returns (ProductListResponse);
    rpc Get(ProductRequest) returns (ProductResponse);
    rpc Create(ProductCreateRequest) returns (ProductCreateResponse);
    rpc Update(ProductUpdateRequest) returns (ProductUpdateResponse);
    rpc Delete(ProductDeleteRequest) returns (ProductDeleteResponse);
}


